cli/src/commands/describe.rs

```diff
@@ -65,39 +76,136 @@ pub(crate) fn cmd_describe(
     args: &DescribeArgs,
 ) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
-    let commit = workspace_command.resolve_single_rev(&args.revision)?;
-    workspace_command.check_rewritable([commit.id()])?;
+    let commits: Vec<_> = workspace_command
+        .parse_union_revsets(&args.revisions)?
+        .evaluate_to_commits()?
+        .try_collect()?; // in reverse topological order
+    if commits.is_empty() {
+        writeln!(ui.status(), "No revisions to describe.")?;
+        return Ok(());
+    }
+    workspace_command.check_rewritable(commits.iter().ids())?;
 
     let mut tx = workspace_command.start_transaction();
-    let mut commit_builder = tx
-        .mut_repo()
-        .rewrite_commit(command.settings(), &commit)
-        .detach();
-    if args.reset_author {
-        commit_builder.set_author(commit_builder.committer().clone());
-    }
+    let tx_description = if commits.len() == 1 {
+        format!("describe commit {}", commits[0].id().hex())
+    } else {
+        format!(
+            "describe commit {} and {} more",
+            commits[0].id().hex(),
+            commits.len() - 1
+        )
+    };
 
-    let description = if args.stdin {
+    let shared_description = if args.stdin {
         let mut buffer = String::new();
         io::stdin().read_to_string(&mut buffer)?;
-        buffer
+        Some(buffer)
     } else if !args.message_paragraphs.is_empty() {
-        join_message_paragraphs(&args.message_paragraphs)
-    } else if args.no_edit {
-        commit.description().to_owned()
+        Some(join_message_paragraphs(&args.message_paragraphs))
     } else {
-        if commit_builder.description().is_empty() {
-            commit_builder.set_description(command.settings().default_description());
+        None
+    };
+
+    let commit_descriptions: HashMap<_, _> = if args.no_edit || shared_description.is_some() {
```

Comment:
Clippy is complaining here with [`mutable-key-type`](https://rust-lang.github.io/rust-clippy/master/index.html#/mutable_key_type) because I'm creating a `HashMap<&Commit, String>`, where presumably `&Commit` can be changed. In this case, `commits` doesn't change so I believe this is a false positive. I'm not sure if there's a better way to write this code or just ignore the Clippy error?

Reply:
Maybe it's false-positive-ish (because `commit.store` has interior mutability, but isn't involved in Eq/Hash), but I think we can easily get around it by using `Vec<(_, _)>` or build `HashMap<&CommitId, _>` upfront. We don't need to filter out unchanged entries here.

Reply:
`Vec<(_, _)>` removes the Clippy error. I guess this is just a (perhaps premature) attempt to optimize by avoiding computing of descendants for unchanged commits. Let me know if I should just remove that.

Reply:
Ah, maybe it avoids rebase of unchanged commits? Inline comment will help here and `transform_descendants()`. (iirc, we use `transform_descendants()` to simplify predecessor chain.)